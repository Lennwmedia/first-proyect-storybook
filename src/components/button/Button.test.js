import * as React from 'react';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

// importar historia por defecto y el resto de historias
import defaultStory, * as stories from './Button.stories';

// crear las posiciones de las historias
const composedStories = composeStories(stories);

// en la descripción de cada historia, se llamara segun el titulo de la historia
describe(`[ ${defaultStory.title}]`, () => {
    // sacar cada llave segun la posición de la historia
    Object.entries(composedStories).forEach(([story, Component]) => {
        it(`should render ${story}`, () => {
            const { asFragment } = render(<Component />);
            expect(asFragment()).toMatchSnapshot();
        })
    })
});