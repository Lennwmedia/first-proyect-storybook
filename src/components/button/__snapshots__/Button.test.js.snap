// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ components/button] should render Primary 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Click me!
  </button>
</DocumentFragment>
`;

exports[`[ components/button] should render Secondary 1`] = `
<DocumentFragment>
  <button
    class="button secondary"
  >
    Click me!
  </button>
</DocumentFragment>
`;

exports[`[ components/button] should render Tertiary 1`] = `
<DocumentFragment>
  <button
    class="button tertiary"
  >
    Click me!
  </button>
</DocumentFragment>
`;
